### PR TITLE
IfcRoad schema documentation

### PR DESCRIPTION
**Pull request by larswik**
Proposal for handling IfcRoad part of #106

### CODE REVIEW

**Comment by christophe-blt** [comment]
Thanks Lars, Jon also told me we should change the name ifcRoad to ifcRoadDomain to be more consistent (same for the other schema). Jon do you validate that ?

**Comment by jmirtsch** [comment]
> 
> 
> Thanks Lars, Jon also told me we should change the name ifcRoad to ifcRoadDomain to be more consistent (same for the other schema). Jon do you validate that ?

Yes, unfortunately being relatively new to the specification baseline,  I didn't note the adjacent naming convention.  This is similar for domains for Rail, Ports and Waterways.  There are also concepts which are contained in the domain that should be moved into resources (this wasn't easily automatically recognized in the initial import from XMI)

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+In-scope road components, elements and equipment:
+
+- Road structure
+- Road guard elements
+- Road sign elements
+- Road paving components
+- Utilities
+  - Lighting, telecom and power
+  - Storm-, surface- water and drainage systems
```
Comment (pjanck) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md:30]: dashes need review

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  - Other underground facilities located in the road body.
+- paved surfaces of:
+  - parking lots;
+  - service areas;
+  - toll plazas;
+  - parking buildings;
+  - ferry ports;
+  - airports.
```
Comment (pjanck) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md:38]: Punctuation is inconsistent.

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+The in scope road types and related spaces have been identified as the most common and widespread ones across the world: 
+
+- Controlled access highway;
+- Dual carriageway;
+- Single carriageway;
+- Street;
+- Bicycle path;
+- Footpath.
+  In-scope junction types:
```
Comment (pjanck) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md:11]: For the sake of readability, I'd include a blank line before and after this line, as well as de-indent it.

**Review by pjanck — APPROVED** [review]
A few remarks about the style. I agree with the content.

**Comment by larswik** [comment]
> > Thanks Lars, Jon also told me we should change the name ifcRoad to ifcRoadDomain to be more consistent (same for the other schema). Jon do you validate that ?
> 
> Yes, unfortunately being relatively new to the specification baseline, I didn't note the adjacent naming convention. This is similar for domains for Rail, Ports and Waterways. There are also concepts which are contained in the domain that should be moved into resources (this wasn't easily automatically recognized in the initial import from XMI)

Yes, I also noticed the difference in naming. Should be IfcRoadDomain. I can do that in the same pull request. I didn't notice however (must be blind) that IfcOpenCrossProfileDef and IfcSectionedSurface were also in the road schema. I didn't check to where, but I guess that IfcRelAssociatesProfileDef should also be moved somewhere. Maybe also the Schema references section needs to be double checked.

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md (hunk 1 above)
Comment (AlexBrad1eyCT) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md:30]:
```suggestion
  - Storm-water, surface-water and drainage systems
```

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+The _IfcRoad_ schema forms part of the Domain Layer of the IFC Model. It extends the ideas concerning infrastructure outlined in the _IfcSharedInfrastructureElements_ schema. It defines concepts specific to road infrastructure.
+
+The in scope road types and related spaces have been identified as the most common and widespread ones across the world: 
+
+- Controlled access highway;
+- Dual carriageway;
+- Single carriageway;
+- Street;
+- Bicycle path;
+- Footpath.
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md:10]:
```suggestion
- Controlled access highway
- Dual carriageway
- Single carriageway
- Street
- Bicycle path
- Footpath
```

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  In-scope junction types:
+- Interchange:
+  - overpass;
+  - underpass;
+  - ramp.
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md:15]:
```suggestion
  - overpass
  - underpass
  - ramp
```

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+- Intersection:
+  - Intersecting roads (3, 4, …, 7 way);
+  - roundabout or traffic circle;
+  - pedestrian crossing;
+  - bicycle crossing.
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md:20]:
```suggestion
  - Intersecting roads (3, 4, …, 7 way)
  - roundabout or traffic circle
  - pedestrian crossing
  - bicycle crossing
```

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md (hunk 2 above)
Comment (AlexBrad1eyCT) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md:38]:
```suggestion
  - parking lots
  - service areas
  - toll plazas
  - parking buildings
  - ferry ports
  - airports
```

**Review by AlexBrad1eyCT — CHANGES_REQUESTED** [review]
Content is great @larswik, just corrected the punctuation.

Easy Rule: if the bullet is a sentance it needs a full stop of its just a 1-3ish work phrase then just leave it. and its already bulletted so doesnt need semi colons

as jon pointed out:

- IfcOpenCrossProfileDef to IfcProfileResource
- IfcSectionedSurface to IfcGeometryResource
- IfcRelAssociatedProfileDef to IfcProductExtension i would think? as its associates a annotations to profiles

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md (hunk 3 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md:11]: resolved by change suggestions

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md (hunk 2 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md:38]: resolved by change suggestions

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md (hunk 1 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoad/Documentation.md:30]: resolved by change suggestions

**Comment by larswik** [comment]
Should we also do the renaming now? IfcRoad => IfcRoadDomain.

**Comment by larswik** [comment]
Is there a way in IfcDoc to do the moving or do I move the folders in the file structure? 
One related issue:
I looked in DocSchema.xml for IfcRoad and it contains the dependency:
```
<DocSchema xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcRoad" UniqueId="92eb3058-d03d-4217-a075-ff181dcad1b2">
	<SchemaRefs>
		<DocSchemaRef Name="IFCGEOMETRYRESOURCE" UniqueId="e3f2a5a2-8b37-4bf0-be78-b76a6cfd3433">
			<Definitions>
				<DocDefinitionRef Name="IfcPointByDistanceExpression" UniqueId="ed7e89ec-9542-4f9e-82b0-aeca1ee1fe3e" />
			</Definitions>
		</DocSchemaRef>
	</SchemaRefs>
</DocSchema>

```
Since this must come from the IfcSectionedSurface definition, this dependency will go away. 

> Content is great @larswik, just corrected the punctuation.
> 
> Easy Rule: if the bullet is a sentance it needs a full stop of its just a 1-3ish work phrase then just leave it. and its already bulletted so doesnt need semi colons
> 
> as jon pointed out:
> 
> * IfcOpenCrossProfileDef to IfcProfileResource
> * IfcSectionedSurface to IfcGeometryResource
> * IfcRelAssociatedProfileDef to IfcProductExtension i would think? as its associates a annotations to profiles

IfcSectionedSurface to IfcGeometricModelResource right? Alongside e.g. IfcSectionedSolidHorizontal.

**Comment by AlexBrad1eyCT** [comment]
@larswik 

I did a similar thing to PortsAndWaterways, i tryed renaming in IfcDoc and it caused an error and lead to unintended changes. 
my advice:
open in VSCode then rename the folder IfcRoad to IfcRoadDomain
you can move the entities by just moving the folders and the dependancy can be removed.

it will generate loads of changes cause of the folder renaming but they will just be filename changes

**Review by AlexBrad1eyCT — APPROVED** [review]

**Pull request merged**